src/OrchardCore/OrchardCore.Abstractions/Json/Dynamic/JsonDynamicObject.cs

```diff
@@ -107,6 +107,26 @@ public bool Remove(string key)
 
         if (jsonNode is JsonValue jsonValue)
         {
+            var valueKind = jsonValue.GetValueKind();
+            switch (valueKind)
+            {
+                case JsonValueKind.String:
+                    if(jsonValue.TryGetValue<DateTime>(out var datetime))
```

Comment:
I've added some type conversion tests here，If this scenario is acceptable, perhaps we need to add some abstraction converters for some custom development scenarios